### PR TITLE
[ISSUE #130] Add Shutdown of PushConsumer

### PR DESCRIPTION
**Pull request by wenfengwang**
## What is the purpose of the change

Add Shutdown of PushConsumer

## Brief changelog

- Add Shutdown of PushConsumer
- Add mqChanged impl of PushConsumer

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: examples/consumer/pull/main.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ package main
 import (
 	"context"
 	"fmt"
+	"github.com/apache/rocketmq-client-go"
```
Comment (xujianhai666) [inline examples/consumer/pull/main.go]: 这个goimports 格式化下哈

File: api.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package rocketmq
 
 import (
 	"context"
+	"github.com/pkg/errors"
```
Comment (xujianhai666) [inline api.go]: goimports 格式化下哈

File: consumer/push_consumer.go [hunk 3]
```diff
@@ -216,7 +216,43 @@ func (pc *pushConsumer) IsUnitMode() bool {
...
+	v, exit := pc.subscriptionDataTable.Load(topic)
+	if !exit {
+		return
+	}
+	data := v.(*internal.SubscriptionData)
+	newVersion := time.Now().UnixNano()
+	rlog.Infof("the MessageQueue changed, also update version: %d to %d", data.SubVersion, newVersion)
+	data.SubVersion = newVersion
+
+	data.SubVersion = time.Now().UnixNano()
```
Comment (xujianhai666) [inline consumer/push_consumer.go]: 是不是代码重复了？

File: api.go (hunk 2 above)
Comment (wenfengwang) [reply api.go]: done

File: consumer/push_consumer.go (hunk 3 above)
Comment (wenfengwang) [reply consumer/push_consumer.go]: done

**Comment by xujianhai666** [comment]
LGTM

File: consumer/push_consumer.go [hunk 4]
```diff
@@ -216,7 +216,41 @@ func (pc *pushConsumer) IsUnitMode() bool {
...
+	v, exit := pc.subscriptionDataTable.Load(topic)
+	if !exit {
+		return
+	}
+	data := v.(*internal.SubscriptionData)
+	newVersion := time.Now().UnixNano()
+	rlog.Infof("the MessageQueue changed, also update version: %d to %d", data.SubVersion, newVersion)
+	data.SubVersion = newVersion
+
+	// TODO: optimize
```
Comment (ShannonDing) [inline consumer/push_consumer.go]: IMO, it better to remove "TODO" here which make the developer confused, and you can create an issue to record that it needs polish in this file here.

Comment (ShannonDing) [reply consumer/push_consumer.go]: @wenfengwang

Comment (wenfengwang) [reply consumer/push_consumer.go]: issue created: https://github.com/apache/rocketmq-client-go/issues/175

**Review by zongtanghu — APPROVED** [review]

**Pull request merged**